Added convenience methods for java primitive types

Peng1104 opened the pull request:
After some time using this library, I have found it s annoying to have to convert myself the Scalar into a java primitive type like booleans, shorts, bytes or BigIntegers (When reading a hex of a color) so i have made this addon for this library where the Scalar interface has all the java primitive type convertions of it String value.

zoeself commented:
@Peng1104 thank you for your Pull Request. I'll assign someone to review it soon. 

 If this PR solves a ``todo`` from the code, please don't forget to remove it.

zoeself commented:
@amihaiemil I couldn't find any assignee for this task. This is either because there are no contributors with role ``REV`` available or because the project does not have enough funds.

Please, make sure there is at least one available contributor with the required role and the project can afford to pay them.

amihaiemil commented:
@Peng1104 Thank you for your Pull Request, but:

* YamlMapping and YamlSequence already have these kind of methods (maybe not all of which you added).
* These methods make no sense on Scalar. You will get a Scalar from a mapping or a sequence, right? If you have a YamlMapping and want to get an integer, you just say ``YamlMapping.integer(...)`` instead of ``YamlMapping.value(...).integer()``.
* Your PR is too big and it also contains a lot of other unrelated changes, including formatting, javadoc changes and refactoring of existing code. 

Please, open an Issue with exactly which methods you would like to add and why, then we can maybe go further. But in any case, I'm going to ask that when you make a PR, you change only what you want to add. A PR containing more than 1 topic of changes is always bad :D

Peng1104 commented:
@amihaiemil, About your concerns,
-  YamlMapping and YamlSequence have some of the methods as you have pointed, I have added a fast way to read booleans, shorts and bytes from them, also added a way to read a decimal number whit a MathContext, so that you have the ability to set a precision and rounding rule for the number. And the last convenince that I have added is to read a integer on another base then 10, like 16 for colors using the BigInteger.
- About these methods in the Scalar Interface, I found it better to put them directly into Scalar so that you have a "unified" way of transforming its value into a primitive, a way to have assurance that the Scalar translation in `YamlMapping.integer(...)` is equals to `YamlSequence.integer(...)` as there are Scalars that are native numbers or booleans.
- I agree whit you that this PR is really Big I will split this PR in some other smaller ones so that it is clear what I have changed
